Sources/ImperialDeviantArt/DeviantArtRouter.swift

```diff
@@ -2,35 +2,38 @@ import Foundation
 import Vapor
 
 struct DeviantArtRouter: FederatedServiceRouter {
+    /// FederatedServiceRouter properties
     let tokens: any FederatedServiceTokens
-    let callbackCompletion: @Sendable (Request, String) async throws -> any AsyncResponseEncodable
-    let scope: [String]
+    let callbackCompletion: @Sendable (Request, String, ByteBuffer?) async throws -> any AsyncResponseEncodable
     let callbackURL: String
     let accessTokenURL: String = "https://www.deviantart.com/oauth2/token"
+    /// Local properties
+    let queryItems: [URLQueryItem]
 
     init(
-        callback: String, scope: [String], completion: @escaping @Sendable (Request, String) async throws -> some AsyncResponseEncodable
+        callback: String, queryItems: [URLQueryItem], completion: @escaping @Sendable (Request, String, ByteBuffer?) async throws -> some AsyncResponseEncodable
     ) throws {
         self.tokens = try DeviantArtAuth()
         self.callbackURL = callback
         self.callbackCompletion = completion
-        self.scope = scope
+        self.queryItems = queryItems + [
```

Comment:
The query items are created in the initializer instead `authURL`

And that's the same basic pattern used in all the providers.